server/controllers/files.py

```diff
@@ -0,0 +1,41 @@
+import logging
+
+import libcloud
+from flask import Blueprint, redirect, Response, abort
+from flask_login import login_required, current_user
+
+from server.extensions import storage
+from server.models import ExternalFile
+
+files = Blueprint('files', __name__)
+
+logger = logging.getLogger(__name__)
+
+@files.route("/files/<hashid:file_id>")
+@login_required
+def file_url(file_id):
+    ext_file = ExternalFile.query.filter_by(id=file_id, deleted=False).first()
+    if not ext_file or not ExternalFile.can(ext_file, current_user, 'download'):
+        abort(404)
+
+    try:
+        storage_obj = ext_file.object
```

Comment:
I find it extremely surprising that accessing a property can raise an exception. We could make it a plain method, or even could catch the exception in the @property and have it return None in that case